functions/__sf_section_golang.fish

```diff
@@ -0,0 +1,43 @@
+#
+# Go
+#
+# Go is an open source programming language that makes it easy
+# to build efficient software.
+# Link: https://golang.org/
+
+function __sf_section_golang -d "Display the current go version if you're inside GOPATH"
+	# ------------------------------------------------------------------------------
+	# Configuration
+	# ------------------------------------------------------------------------------
+
+	__sf_util_set_default SPACEFISH_GOLANG_SHOW true
+	__sf_util_set_default SPACEFISH_GOLANG_PREFIX $SPACEFISH_PROMPT_DEFAULT_PREFIX
+	__sf_util_set_default SPACEFISH_GOLANG_SUFFIX $SPACEFISH_PROMPT_DEFAULT_SUFFIX
+	__sf_util_set_default SPACEFISH_GOLANG_SYMBOL "🐹 "
+	__sf_util_set_default SPACEFISH_GOLANG_COLOR cyan
+
+	# ------------------------------------------------------------------------------
+	# Section
+	# ------------------------------------------------------------------------------
+
+	[ $SPACEFISH_GOLANG_SHOW = false ]; and return
+
+	if not test -d Godeps \
+		-o -f glide.yaml \
+		-o (count *.go) -gt 0 \
+		-o ([ -n $GOPATH ]; and string match $GOPATH $PWD)
+		return
+	end
+
+	if type -q go
+		set -g sf_go_version (go version | awk '{ print substr($3, 3) }')
```

Comment:
Sounds good! I was basing it largely off the NodeJS section. But it makes sense since we're not using the variable yet to make it local.

Reply:
The reason for making it global in the Node section is so that we could cache the version based on the nvm binary directory since the built-in nvm version command is extremely slow.